shared/node.go

```diff
@@ -23,28 +23,28 @@ type Node struct {
 	Address cryptoPocket.Address
 }
 
-func Create(cfg *genesis.Config, genesis *genesis.GenesisState) (n *Node, err error) {
-	persistenceMod, err := persistence.Create(cfg, genesis)
+func Create(cfg, genesis map[string]json.RawMessage) (n *Node, err error) {
+	persistenceMod, err := persistence.Create(cfg["Persistence"], genesis["PersistenceGenesisState"])
 	if err != nil {
 		return nil, err
 	}
 
-	p2pMod, err := p2p.Create(cfg, genesis)
+	p2pMod, err := p2p.Create(cfg["P2P"], genesis["P2PGenesisState"])
 	if err != nil {
 		return nil, err
 	}
 
-	utilityMod, err := utility.Create(cfg, genesis)
+	utilityMod, err := utility.Create(cfg["Utility"], genesis["UtilityGenesisState"])
 	if err != nil {
 		return nil, err
 	}
 
-	consensusMod, err := consensus.Create(cfg, genesis)
+	consensusMod, err := consensus.Create(cfg["Consensus"], genesis["ConsensusGenesisState"])
 	if err != nil {
 		return nil, err
 	}
 
-	telemetryMod, err := telemetry.Create(cfg, genesis)
+	telemetryMod, err := telemetry.Create(cfg["Telemetry"], genesis["TelemetryGenesisState"]) // TODO (team; discuss) is telemetry a proper module or not?
```

Comment:
Remove `// TODO (team; discuss) is telemetry a proper module or not?`

The goal of the app-specific bus is to add the ability to add different types of modules (not limiting ourselves to just the 4 in perpetuity). The easy with which we were able to add `node.GetBus().GetTelemtry().Emit()` reaffirmed the design decision, and I see us adding others in the future as well.